### PR TITLE
[233] Small Terraform Reorganization

### PR DESCRIPTION
**Pull request by jaronoff97**
This PR is some small clean up for terraform (mostly imports and references here and there.) It doesn't block on the public OSS stuff. Part of this PR was figuring out how to run terraform locally, this will allow us to upgrade to terraform 1.0.1 with some more ease, as well as do some refactoring of our modules to clean up.

### CODE REVIEW

File: terraform/remote.tf
Comment (Mr0grog) [inline terraform/remote.tf:9]: Now that this is here, what can or should we do to discourage running commands that might taint the production state? Do they always require manually locking the workspace, and is that clear enough?

(In reference to Slack discussion: https://usdigitalresponse.slack.com/archives/C01S4JWSMT7/p1625088086008200?thread_ts=1625024517.032900&cid=C01S4JWSMT7)

File: terraform/variables.tf
Comment (Mr0grog) [inline terraform/variables.tf:120]: Are these in preparation for a bastion config that does not actually exist on this PR yet? I don’t see them actually referenced anywhere.

File: terraform/variables.tf
Comment (Mr0grog) [inline terraform/variables.tf:146]: Necessary for local `terraform` to import/plan/apply, or is this just cleaning up the warning we currently get in all our plans in Terraform Cloud?

**Review by Mr0grog — COMMENTED** [review]
Getting back around to this after all the mess from Monday night.

What’s the current plan for how the work here needs to continue? I *think* the changes to production state don’t have to happen until right before this merges, correct? If so, I think you should:
1. Hold off on those until you merge this.
2. Describe in this PR the Terraform commands that need to be run alongside/before merging.
3. Like we talked about on Slack, work out a plan and time window for doing the above and articulate it in this PR.

File: terraform/variables.tf
Comment (jaronoff97) [reply terraform/variables.tf:146]: Correct, this is just to clean up that warning which was preventing my local planning

File: terraform/variables.tf
Comment (jaronoff97) [reply terraform/variables.tf:120]: ah yeah this was when i was attempting to import the ec2 instance in to this, i should remove it.

**Review by astonm — APPROVED** [review]
Nice, all looks pretty good to me. One weird thing is that the plan seems to do something weird with the main security group:

```terraform
      ~ ingress                = [
          - {
              - cidr_blocks      = []
              - description      = ""
              - from_port        = 5432
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "tcp"
              - security_groups  = [
                  - "sg-06fc404762de692db",
                  - "sg-0f04623ee5e2a8214",
                ]
              - self             = false
              - to_port          = 5432
            },
          - {
              - cidr_blocks      = []
              - description      = ""
              - from_port        = 5432
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "tcp"
              - security_groups  = []
              - self             = false
              - to_port          = 5432
            },
          + {
              + cidr_blocks      = []
              + description      = null
              + from_port        = 5432
              + ipv6_cidr_blocks = []
              + prefix_list_ids  = []
              + protocol         = "tcp"
              + security_groups  = []
              + self             = false
              + to_port          = 5432
            },
            # (1 unchanged element hidden)
        ]
```

Hopefully if you merge `main` into this branch, that will clean that part up. And then, as discussed offline, make sure you create the bastion security group before merging this PR.

**Pull request merged**
